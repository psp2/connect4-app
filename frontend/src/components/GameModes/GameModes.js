import React from "react";
import "../../pages/Menus/Menus.scss";

function GameModes(props) {
  function selectGameMode(mode) {
    props.changeGameMode(mode);
  }

  return (
    <div className="MenuBoard">
      <div className="MenuTitle">
        <div className="MenuTitleWhite">Game</div>
        <div className="MenuTitleRed">Modes</div>
      </div>
      <div className="MenuRow">
        <div className="WhiteButton">
          <div className="ButtonCaption" onClick={() => selectGameMode(1)}>
            Connect 4
          </div>
        </div>
        {props.numPlayers === 1 && (
          <div className="YellowButton" onClick={() => selectGameMode(2)}>
            <div className="ButtonCaption">Don't Connect 4</div>
          </div>
        )}
        {props.numPlayers !== 1 && (
          <div className="GreenButton" onClick={() => selectGameMode(2)}>
            <div className="ButtonCaption">Don't Connect 4</div>
          </div>
        )}
        <div className="RedButton" onClick={() => selectGameMode(3)}>
          <div className="ButtonCaption">Mayhem</div>
        </div>
      </div>
      {props.numPlayers === 1 && (
        <div className="BackButtonYellow" onClick={() => props.resetPlayers(0)}>
          <div className="ButtonCaption">Go Back</div>
        </div>
      )}
      {props.numPlayers !== 1 && (
        <div className="BackButtonGreen" onClick={() => props.resetPlayers(0)}>
          <div className="ButtonCaption">Go Back</div>
        </div>
      )}
    </div>
  );
}

export default GameModes;
