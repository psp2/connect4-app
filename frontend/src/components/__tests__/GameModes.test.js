import React from "react";
import GameModes from "../GameModes/GameModes";
import renderer from "react-test-renderer";

it("GameModes, renders correctly", () => {
  const tree = renderer.create(<GameModes />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("GameModes, single player", () => {
  // should have yellow middle buttons
  const tree = renderer.create(<GameModes numPlayers={1} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("GameModes, multi player", () => {
  // should have green middle buttons
  const tree = renderer.create(<GameModes numPlayers={2} />).toJSON();
  expect(tree).toMatchSnapshot();
});
