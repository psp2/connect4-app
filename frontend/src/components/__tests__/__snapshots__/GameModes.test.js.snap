// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameModes, multi player 1`] = `
<div
  className="MenuBoard"
>
  <div
    className="MenuTitle"
  >
    <div
      className="MenuTitleWhite"
    >
      Game
    </div>
    <div
      className="MenuTitleRed"
    >
      Modes
    </div>
  </div>
  <div
    className="MenuRow"
  >
    <div
      className="WhiteButton"
    >
      <div
        className="ButtonCaption"
        onClick={[Function]}
      >
        Connect 4
      </div>
    </div>
    <div
      className="GreenButton"
      onClick={[Function]}
    >
      <div
        className="ButtonCaption"
      >
        Don't Connect 4
      </div>
    </div>
    <div
      className="RedButton"
      onClick={[Function]}
    >
      <div
        className="ButtonCaption"
      >
        Mayhem
      </div>
    </div>
  </div>
  <div
    className="MenuButtonRow"
  >
    <div
      className="BackButtonGreen"
      onClick={[Function]}
    >
      <div
        className="ButtonCaption"
      >
        Back
      </div>
    </div>
  </div>
</div>
`;

exports[`GameModes, renders correctly 1`] = `
<div
  className="MenuBoard"
>
  <div
    className="MenuTitle"
  >
    <div
      className="MenuTitleWhite"
    >
      Game
    </div>
    <div
      className="MenuTitleRed"
    >
      Modes
    </div>
  </div>
  <div
    className="MenuRow"
  >
    <div
      className="WhiteButton"
    >
      <div
        className="ButtonCaption"
        onClick={[Function]}
      >
        Connect 4
      </div>
    </div>
    <div
      className="GreenButton"
      onClick={[Function]}
    >
      <div
        className="ButtonCaption"
      >
        Don't Connect 4
      </div>
    </div>
    <div
      className="RedButton"
      onClick={[Function]}
    >
      <div
        className="ButtonCaption"
      >
        Mayhem
      </div>
    </div>
  </div>
  <div
    className="MenuButtonRow"
  >
    <div
      className="BackButtonGreen"
      onClick={[Function]}
    >
      <div
        className="ButtonCaption"
      >
        Back
      </div>
    </div>
  </div>
</div>
`;

exports[`GameModes, single player 1`] = `
<div
  className="MenuBoard"
>
  <div
    className="MenuTitle"
  >
    <div
      className="MenuTitleWhite"
    >
      Game
    </div>
    <div
      className="MenuTitleRed"
    >
      Modes
    </div>
  </div>
  <div
    className="MenuRow"
  >
    <div
      className="WhiteButton"
    >
      <div
        className="ButtonCaption"
        onClick={[Function]}
      >
        Connect 4
      </div>
    </div>
    <div
      className="YellowButton"
      onClick={[Function]}
    >
      <div
        className="ButtonCaption"
      >
        Don't Connect 4
      </div>
    </div>
    <div
      className="RedButton"
      onClick={[Function]}
    >
      <div
        className="ButtonCaption"
      >
        Mayhem
      </div>
    </div>
  </div>
  <div
    className="MenuButtonRow"
  >
    <div
      className="BackButtonYellow"
      onClick={[Function]}
    >
      <div
        className="ButtonCaption"
      >
        Back
      </div>
    </div>
  </div>
</div>
`;
